tests/test-coauthors-plus.php

```diff
@@ -679,6 +699,1434 @@ public function test_update_author_term_when_author_term_not_exist() {
+	 * Same as above, except not the linked user is passed up front. This should result in
+	 * the wp_posts.post_author column being set to the linked user, and there should be
+	 * 3 WP_Terms for the authors.
+	 *
+	 * @return void
+	 */
+	public function test_assign_multiple_post_authors_one_linked_passed_first() {
+		$random_username = 'random_user_' . rand( 1, 1000 );
+		$display_name = str_replace( '_', ' ', $random_username );
+
+		$guest_author_1_id = $this->_cap->guest_authors->create(
+			array(
+				'user_login'   => $random_username,
+				'display_name' => $display_name,
+			)
+		);
+		$guest_author_1 = $this->_cap->get_coauthor_by( 'id', $guest_author_1_id );
+
+		$this->assertIsObject( $guest_author_1 );
+		$this->assertThat(
+			$guest_author_1,
+			$this->logicalNot(
+				$this->isInstanceOf( WP_User::class )
+			)
+		);
+
+		$this->_cap->guest_authors->create_guest_author_from_user_id( $this->author3->ID );
+
+		$linked_author_1 = $this->_cap->get_coauthor_by( 'id', $this->author3->ID );
+		$this->assertIsObject( $linked_author_1 );
+		$this->assertThat(
+			$linked_author_1,
+			$this->logicalNot(
+				$this->isInstanceOf( WP_User::class )
+			)
+		);
+		$this->assertTrue( property_exists( $linked_author_1, 'is_wp_user' ) );
+		$this->assertTrue( $linked_author_1->is_wp_user );
+		$this->assertTrue( property_exists( $linked_author_1, 'wp_user' ) );
+		$this->assertInstanceOf( WP_User::class, $linked_author_1->wp_user );
+		$this->assertEquals( $this->author3->ID, $linked_author_1->wp_user->ID );
+
+		$post_id = $this->factory()->post->create(
+			array(
+				'post_author' => $this->editor1->ID,
+				'post_status' => 'publish',
+				'post_type'   => 'post',
+			)
+		);
+
+		$query = new WP_Query(
+			array(
+				'p' => $post_id,
+			)
+		);
+
+		$this->assertEquals( 1, $query->found_posts );
+		$this->assertEquals( $this->editor1->ID, $query->posts[0]->post_author );
+
+		$result = $this->_cap->add_coauthors(
+			$post_id,
+			array(
+				$linked_author_1->user_login,
+				$this->author2->user_login,
+				$guest_author_1->user_login,
+			)
+		);
+
+		$this->assertTrue( $result );
+
+		$second_query = new WP_Query(
+			array(
+				'p' => $post_id,
+			)
+		);
+
+		$this->assertEquals( 1, $second_query->found_posts );
+		$this->assertEquals( $this->author3->ID, $second_query->posts[0]->post_author );
+
+		$post_author_terms = wp_get_post_terms( $post_id, $this->_cap->coauthor_taxonomy );
+
+		$this->assertTrue( is_array( $post_author_terms ) );
+		$this->assertCount( 3, $post_author_terms );
+
+		$author_slugs = array(
+			'cap-' . $this->author2->user_login,
+			'cap-' . $this->author3->user_login,
+			'cap-' . $guest_author_1->user_login,
+		);
+
+		foreach ( $post_author_terms as $term ) {
+			$this->assertInstanceOf( WP_Term::class, $term );
+			$this->assertContains( $term->slug, $author_slugs );
+		}
+	}
+
+	/**
+	 * Here we are testing a similar scenario as above, except we are passing the linked user's WP_User login field
```

Comment:
If the test is to check that a user_login field is correctly converted, and applied, then you only need two users at most, and it can be simplified down.

Reply:
The reason I added 3 authors is to ensure and signify that the functionality behaves correctly, as expected, no matter how many authors are passed during assignment. This ensures that, in the future, should a code change come along that uses `array_shift` or `array_pop`, this test would fail, because the function is not picking up the correct author for linkage.